infra/Pos/Slotting/Util.hs

```diff
@@ -124,20 +123,20 @@ onNewSlotImpl
     :: forall ctx m. OnNewSlot ctx m
     => Bool -> Bool -> (SlotId -> m ()) -> m ()
 onNewSlotImpl withLogging startImmediately action =
-    reportingFatal impl `catch` workerHandler
+    impl `catch` workerHandler
   where
     impl = onNewSlotDo withLogging Nothing startImmediately actionWithCatch
+    -- [CSL-1578] FIXME: it is awful!!!
```

Comment:
Sorry, what exactly is awful here?

Reply:
Hm, probably I overreacted :D
The bad think is `fork` which is applied to each invocation of action (on each tick) which forces us to catch exceptions from action separately. [This comment](https://github.com/input-output-hk/cardano-sl/pull/1163#issuecomment-318136810) provides detailed overview of consequences of having `fork` there.
Action itself should determine whether it wants to be fork (one can pass `fork action` instead of `action` if it's necessary).
Probably catching exceptions from `action` separately is not that bad. So ok, I will remove this comment and will use `reportOrLogE` here (especially because nobody knows when CSL-1578 will be done).